graphene_sqlalchemy/converter.py

```diff
@@ -207,97 +214,181 @@ def inner(fn):
+@convert_sqlalchemy_type.register(column_type_eq(sqa_utils.UUIDType))
+@convert_sqlalchemy_type.register(column_type_eq(uuid.UUID))
+def convert_column_to_uuid(
+    type_arg: Any,
+    **kwargs,
+):
     return graphene.UUID
 
 
-@convert_sqlalchemy_type.register(sqa_types.DateTime)
-def convert_column_to_datetime(type, column, registry=None):
+@convert_sqlalchemy_type.register(column_type_eq(sqa_types.DateTime))
+@convert_sqlalchemy_type.register(column_type_eq(datetime.datetime))
+def convert_column_to_datetime(
+    type_arg: Any,
+    **kwargs,
+):
     return graphene.DateTime
 
 
-@convert_sqlalchemy_type.register(sqa_types.Time)
-def convert_column_to_time(type, column, registry=None):
+@convert_sqlalchemy_type.register(column_type_eq(sqa_types.Time))
+@convert_sqlalchemy_type.register(column_type_eq(datetime.time))
+def convert_column_to_time(
+    type_arg: Any,
+    **kwargs,
+):
     return graphene.Time
 
 
-@convert_sqlalchemy_type.register(sqa_types.Date)
-def convert_column_to_date(type, column, registry=None):
+@convert_sqlalchemy_type.register(column_type_eq(sqa_types.Date))
+@convert_sqlalchemy_type.register(column_type_eq(datetime.date))
+def convert_column_to_date(
+    type_arg: Any,
+    **kwargs,
+):
     return graphene.Date
 
 
-@convert_sqlalchemy_type.register(sqa_types.SmallInteger)
-@convert_sqlalchemy_type.register(sqa_types.Integer)
-def convert_column_to_int_or_id(type, column, registry=None):
-    return graphene.ID if column.primary_key else graphene.Int
+@convert_sqlalchemy_type.register(column_type_eq(sqa_types.SmallInteger))
+@convert_sqlalchemy_type.register(column_type_eq(sqa_types.Integer))
+@convert_sqlalchemy_type.register(column_type_eq(int))
+def convert_column_to_int_or_id(
+    type_arg: Any,
+    column: Optional[Union[MapperProperty, hybrid_property]] = None,
+    registry: Registry = None,
+    **kwargs,
+):
+    # fixme drop the primary key processing from here in another pr
+    if column is not None:
+        if getattr(column, "primary_key", False) is True:
+            return graphene.ID
+    return graphene.Int
 
 
-@convert_sqlalchemy_type.register(sqa_types.Boolean)
-def convert_column_to_boolean(type, column, registry=None):
+@convert_sqlalchemy_type.register(column_type_eq(sqa_types.Boolean))
+@convert_sqlalchemy_type.register(column_type_eq(bool))
+def convert_column_to_boolean(
+    type_arg: Any,
+    **kwargs,
+):
     return graphene.Boolean
 
 
-@convert_sqlalchemy_type.register(sqa_types.Float)
-@convert_sqlalchemy_type.register(sqa_types.Numeric)
-@convert_sqlalchemy_type.register(sqa_types.BigInteger)
-def convert_column_to_float(type, column, registry=None):
+@convert_sqlalchemy_type.register(column_type_eq(float))
+@convert_sqlalchemy_type.register(column_type_eq(sqa_types.Float))
+@convert_sqlalchemy_type.register(column_type_eq(sqa_types.Numeric))
+@convert_sqlalchemy_type.register(column_type_eq(sqa_types.BigInteger))
```

Comment:
maybe a bit off topic here, but wondering about the choice of BigInteger->Float. assume this would break down after 64 bits. any plans for graphql-core to provide a [BigInt type like this](https://www.npmjs.com/package/graphql-bigint)?

Reply:
Good point, let's put this into an issue. Will have a look at it later.

Reply:
Started #373